shared/lib_irradproc.h

```diff
@@ -95,11 +95,791 @@ struct poaDecompReq;
 * \param[out] sunn[7] true solar time (hrs)
 * \param[out] sunn[8] extraterrestrial solar irradiance on horizontal at particular time (W/m2)
 */
-void solarpos(int year,int month,int day,int hour,double minute,double lat,double lng,double tz,double sunn[9]);
+void solarpos(int year, int month, int day, int hour, double minute, double lat, double lng, double tz, double sunn[9]);
+
+/**
+*   limit_degrees function calculates the degree value of a parameter within 0-360°
+*   This function is based on the C code implementation of the Solar Position Algorithm
+*   based on a paper by Reda and Andreas published in NREL/TP-
+*	560-34302, 2008.
```

Comment:
I think rather than repeating these lines for each solarpos_spa function, we should divide all the functions in this entire file into groups: old `solarpos`, new `solarpos_spa`, and maybe one for the different sky models (ISOTROPIC, HDKR, PEREZ). The rest can be ungrouped, or grouped some way that makes sense to you.

The commenting convention in this file follows Doxygen. As you can see in the Example on this [docs page](https://www.doxygen.nl/manual/grouping.html), you can make groups of functions using:

```
/** @defgroup group1 The First Group
 *  This is the first group
 *  @{
 */
 
/** @brief class C1 in group 1 */
class C1 {};
 
/** @brief class C2 in group 1 */
class C2 {};
 
/** function in group 1 */
void func() {}
 
/** @} */ // end of group1
```

Which then helps sort it out, as in the resulting [html documentation](https://www.doxygen.nl/manual/examples/group/html/modules.html). This way we can cite the reference once in the group description.